### PR TITLE
contributing: add sections and update phrasing

### PR DESCRIPTION
**Pull request by alexakreizinger**
this PR updates the Fluent Bit contributor guide.

### CODE REVIEW

**Comment by esmerel** [comment]
We use the [rewrap extension for vscode](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) and set 
 the **wrapping column** to 85 or so.

**Comment by lecaros** [comment]
> figure out why the DCO check is failing even though I configured my GPG key :(

You need to configure your git user, then sign off the commit, i.e. `commit -s -m "commit message"`

**Comment by alexakreizinger** [comment]
> > figure out why the DCO check is failing even though I configured my GPG key :(
> 
> You need to configure your git user, then sign off the commit, i.e. `commit -s -m "commit message"`

@lecaros I did configure my key and sign off, and then configured VS Code to automatically sign off all commits for me, but DCO threw an error anyway...

image

I'm not sure if it's because I didn't set a "Signed off by..." message when I configured the key. is there a specific sign-off message DCO is expecting?

**Comment by lecaros** [comment]
Yes, it's expecting the `Signed-off-by... ` message. It's attached when you add the `-s` to your commit.
Screenshot 2024-05-03 at 11 15 11 AM

**Comment by alexakreizinger** [comment]
gotcha. let me try reconfiguring the auto-signoff before I push further commits and see if that yields the correct behavior

**Comment by alexakreizinger** [comment]
@esmerel I think we should hold off on adding the rewrap info until we consult with the Fluent Bit team and Craig; if we do want to add that as a best practice, I think it would make sense to do that at the same time as we add the Vale tests. (I also feel strongly that we shouldn't include info about tag/label types in this guide since GitHub already surfaces definitions for those on the labels page itself.)

@lecaros I believe I successfully configured my signoff key, but when I attempted to rebase and add signoff to the failing commits, I still wasn't able to get DCO approval; is it okay to use the "set DCO to pass" override, or do I need to take other steps to resolve this?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,76 +1,118 @@
...
+This repository contains the files for the
+[Fluent Bit documentation library](https://docs.fluentbit.io/). Keeping these docs
+separate from the [main Fluent Bit repository](https://github.com/fluent/fluent-bit)
+helps reduce the number of commits to the Fluent Bit source code and makes it
+easier to maintain both projects.
+
+Fluent Bit has a group of dedicated maintainers who oversee this repository,
+including several technical writers. These writers will review any pull requests
+you open, so don't be afraid to contribute—even if you're not a writer by trade,
+your suggestions are valuable, and we'll help you wrangle any stray commas.
```
Comment (esmerel) [inline CONTRIBUTING.md:16]:
```suggestion
you open, so don't be afraid to contribute—even if you're not a writer by trade.
Your suggestions are valuable, and we'll help you wrangle any stray commas.
```

little less run-on this way :D

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,76 +1,118 @@
...
 
-Documentation source code structure depends on Fluent Bit source code structure and its versions. In Fluent Bit source code, we  have a stable branch and a development branch; as of now, these are:
+### Stale pull requests
 
-- stable branch: [1.8](https://github.com/fluent/fluent-bit/tree/1.8)
-- development branch: [master](https://github.com/fluent/fluent-bit/tree/master)
+If you open a pull request that requires ongoing discussion or review, the
+Fluent Bit maintainers will add a [`waiting-for-user` tag](#tags) to your pull
+request. This tag means that we're blocked from moving forward until you reply;
+to keep contributions from going stale, we'll wait [three weeks?] for your response,
```
Comment (esmerel) [inline CONTRIBUTING.md:45]:
```suggestion
request. This tag means that we're blocked from moving forward until you reply.
To keep contributions from going stale, we'll wait [three weeks?] for your response,
```

You'll want to get that three weeks more concrete though, we don't want to merge it with a guess.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,76 +1,118 @@
...
-
-Usually, a PR can have multiple commits, but we enforce that every commit only touches one file or interface (we apply the same practice in Fluent Bit source code).
+For more information, refer to GitHub's guide to
+[setting your commit email address in Git](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git).
 
 ### Sign off your commits
 
-Your commits must be **sign off**; this certifies who is the author of the commit. It might sound a bit redundant, but it is needed. If you don't sign-off your commits, our CI system will flag the PR with a [DCO](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) error and the PR will be blocked.
+You must sign off your commits to certify your identity as the commit author. If
+you don't sign off your commits, our CI system will flat the pull request with a
```
Comment (esmerel) [inline CONTRIBUTING.md:91]:
```suggestion
you don't sign off your commits, our CI system will flag the pull request with a
```

**Review by esmerel — APPROVED** [review]
I think this is generally fine - the couple of things that need updating aren't block-worthy.

definitely do fix the time estimate before merging though :D

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexakreizinger) [reply CONTRIBUTING.md:16]: good suggestion, I like yours better 👍

File: CONTRIBUTING.md (hunk 2 above)
Comment (alexakreizinger) [reply CONTRIBUTING.md:45]: yep, good call. per our Slack discussion I'm gonna bump this to six weeks (but in a separate commit so I can modify a few more lines at once)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,76 +1,118 @@
...
-Documentation source code structure depends on Fluent Bit source code structure and its versions. In Fluent Bit source code, we  have a stable branch and a development branch; as of now, these are:
+### Stale pull requests
 
-- stable branch: [1.8](https://github.com/fluent/fluent-bit/tree/1.8)
-- development branch: [master](https://github.com/fluent/fluent-bit/tree/master)
+If you open a pull request that requires ongoing discussion or review, the
+Fluent Bit maintainers will add a [`waiting-for-user` tag](#tags) to your pull
+request. This tag means that we're blocked from moving forward until you reply;
+to keep contributions from going stale, we'll wait [three weeks?] for your response,
+and close the pull request if we don't hear back from you.
```
Comment (alexakreizinger) [inline CONTRIBUTING.md:46]:
```suggestion
request. This tag means that we're blocked from moving forward until you reply;
to keep contributions from going stale, we'll wait 45 days for your response,
but we may close the pull request if we don't hear back from you by then.
```

**Pull request merged**